### PR TITLE
Containers for elements

### PR DESCRIPTION
**Pull request by akaDuality**
# Infrastructure
- [x] Add saving to file
- [x] Use common code for file decoding/encoding
- [x] Make [custom json with container](https://github.com/VODGroup/VoiceOverSamples/tree/main/Ru/Dodo%20Pizza/Dodo%20Product%20card.vodesign)
- [x] Draw the sample container on left panel
- [x] Fix Preview's build
- [x] Fix tests in project
- [x] Draw an element that are placed inside container

# Feature
- [ ] Use [NSTreeController](https://developer.apple.com/documentation/appkit/nstreecontroller)
- [ ] Fix: do not rearrange items after selection
- [ ] Rework cell to custom subclass and handle selection appearance in cell
- [ ] Add multiple selection to tree panel
- [ ] Add action to wrap in container
- [ ] Rename package TextUI to TextTree?

### CODE REVIEW

File: VoiceOver Designer/Features/Sources/Editor/EditorPresenter.swift [hunk 1]
```diff
@@ -72,7 +72,11 @@ public class EditorPresenter: DocumentPresenter {
         self.screenUI = screenUI
         self.drawingController = DrawingController(view: ui)
         
-        draw(controls: document.controls)
+        let controls = document.controls.compactMap({
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Editor/EditorPresenter.swift:75]: Вынести этот compactMap в общий код, чтобы не надо было дублировать

File: VoiceOver Designer/Features/Sources/Editor/EditorPresenter.swift [hunk 2]
```diff
@@ -81,6 +85,9 @@ public class EditorPresenter: DocumentPresenter {
     private func redrawOnControlChanges() {
         document
             .controlsPublisher
+            .map({
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Editor/EditorPresenter.swift:88]: Вот тут тот же compactMap

File: Shared/Sources/Document/Documents/AccessibilityView+ExtractElements.swift [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+extension Array where Element == any AccessibilityView {
```
Comment (drucelweisse) [inline Shared/Sources/Document/Documents/AccessibilityView+ExtractElements.swift:3]: cool!

File: Shared/Sources/Document/Documents/DocumentSaveService.swift [hunk 4]
```diff
@@ -7,27 +7,57 @@
 
 import Foundation
 
-class DocumentSaveService {
+protocol DataProvier {
```
Comment (drucelweisse) [inline Shared/Sources/Document/Documents/DocumentSaveService.swift:10]: typo in DataProvider

**Review by drucelweisse — APPROVED** [review]
Just one typo

**Pull request merged**